students/philip_behrend/lesson07/assignment/parallel.py

```diff
@@ -143,23 +140,22 @@ def show_rentals(product_id):
 
 
 if __name__ == '__main__':
-    t1 = import_file('products.csv')
-    out_data = Queue()
-    prod_thread = Thread(target=import_file, args=(out_data, 'products.csv',0))
-    prod_thread.start()
-    cust_thread = Thread(target=import_file, args=(out_data, 'customers.csv',1))
-    cust_thread.start()
-    rent_thread = Thread(target=import_file, args=(out_data, 'rentals.csv',2))
-    rent_thread.start()
-
-    prod_thread.join()
-    cust_thread.join()
-    rent_thread.join()
-    prod_output = out_data.get()
```

Comment:
Hi Philip,

There's one issue with popping results out of the _out_data_ queue in this way. You're assuming the results will always go into the queue in the same order, which is not guaranteed. Whichever thread finishes first will be the one pushing results into the queue and that could be any of the three threads. Think of a different way of figuring out which thread the result popping out of the queue comes from. 

Also, you can use _while not queue.empty():_ to iterate through the queue, although doing it with three code lines as you're doing it here is also fine.

-Luis